l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py

```diff
@@ -697,120 +694,96 @@ def setAttachments(self, cr, uid, invoice, body, context=None):
+        invoices = invoice_model.browse(invoice_ids)
 
         for invoice in invoices:
             if not partner:
                 partner = invoice.partner_id
             if invoice.partner_id != partner:
-                raise orm.except_orm(
-                    _('Error!'),
+                raise UserError(
                     _('Invoices must belong to the same partner'))
 
         return partner
 
-    def exportFatturaPA(self, cr, uid, ids, context=None):
-        if context is None:
-            context = {}
+    @api.multi
+    def exportFatturaPA(self):
 
         # self.setNameSpace()
 
-        model_data_obj = self.pool['ir.model.data']
-        invoice_obj = self.pool['account.invoice']
+        model_data_obj = self.env['ir.model.data']
+        invoice_obj = self.env['account.invoice']
+        invoice_ids = self.env.context.get('active_ids', False)
+        partner = self.getPartnerId(invoice_ids)
 
-        self.fatturapa = FatturaElettronica(versione='FPA12')
-        invoice_ids = context.get('active_ids', False)
-        partner = self.getPartnerId(cr, uid, invoice_ids, context=context)
+        if partner.is_pa:
+            fatturapa = FatturaElettronica(versione='FPA12')
+        else:
+            fatturapa = FatturaElettronica(versione='FPR12')
 
-        user_obj = self.pool['res.users']
-        company = user_obj.browse(cr, uid, uid).company_id
-        context_partner = context.copy()
+        company = self.env.user.company_id
+        context_partner = self.env.context.copy()
         context_partner.update({'lang': partner.lang})
         try:
-            self.setFatturaElettronicaHeader(cr, uid, company,
-                                             partner, context=context_partner)
+            self.with_context(context_partner).setFatturaElettronicaHeader(
+                company, partner, fatturapa)
             for invoice_id in invoice_ids:
-                inv = invoice_obj.browse(
-                    cr, uid, invoice_id, context=context_partner)
+                inv = invoice_obj.with_context(context_partner).browse(
+                    invoice_id)
                 if inv.fatturapa_attachment_out_id:
-                    raise orm.except_orm(
-                        _("Error"),
+                    raise UserError(
                         _("Invoice %s has FatturaPA Export File yet") % (
                             inv.number))
                 invoice_body = FatturaElettronicaBodyType()
-                self.setFatturaElettronicaBody(
-                    cr, uid, inv, invoice_body, context=context_partner)
-                self.fatturapa.FatturaElettronicaBody.append(invoice_body)
+                inv.preventive_checks()
+                self.with_context(context_partner).setFatturaElettronicaBody(
+                    inv, invoice_body)
+                fatturapa.FatturaElettronicaBody.append(invoice_body)
                 # TODO DatiVeicoli
 
```

Comment:
in caso la sequenza sia pluriennale, a me risulta serva passare il fiscalyear_id nel context, con una cosa simile:
`            invoices = invoice_obj.browse(invoice_ids)
            fy_id = invoices.mapped('period_id.fiscalyear_id.id')
            if len(fy_id) > 1:
                raise UserError(_('More than 1 fy included in the file!'))
            number = self.with_context(
                fiscalyear_id=fy_id[0]
            ).setProgressivoInvio(fatturapa)`

Reply:
chiudo questa segnalazione, come sulle altre versioni la sequenza non è prevista per essere pluriennale